src/main/kotlin/it/pagopa/wallet/services/PaymentMethodsService.kt

```diff
@@ -37,19 +48,28 @@ class PaymentMethodsService(
     private fun retrievePaymentMethodByApi(paymentMethodId: String): Mono<PaymentMethodResponse> =
         ecommercePaymentMethodsClient
             .getPaymentMethodById(paymentMethodId)
-            .flatMap { savePaymentMethodIntoCache(it) }
+            .doOnSuccess {
+                val emitResult = paymentMethodCacheSaveSink.tryEmitNext(it)
+                logger.debug("Emit paymentMethodCacheSaveSink result: {}", emitResult)
+            }
             .doOnError { logger.error("Error during call to payment method: [$paymentMethodId]") }
 
-    private fun savePaymentMethodIntoCache(paymentMethodResponse: PaymentMethodResponse) =
-        mono(Dispatchers.IO) {
-                logger.debug("Save payment method into cache: [${paymentMethodResponse.id}]")
-                paymentMethodsRedisTemplate.save(paymentMethodResponse)
-            }
-            .onErrorResume {
-                logger.error(
-                    "Error saving payment method into cache: [${paymentMethodResponse.id}]"
-                )
-                Mono.just(Unit)
+    fun subscribePaymentMethodCacheSaveSink(): Disposable =
+        paymentMethodCacheSaveSink
+            .asFlux()
+            .flatMap { paymentMethodResponse ->
+                mono(Dispatchers.IO) {
+                        logger.debug(
+                            "Save payment method into cache: [${paymentMethodResponse.id}]"
+                        )
+                        paymentMethodsRedisTemplate.save(paymentMethodResponse)
+                    }
+                    .doOnError {
+                        logger.error(
+                            "Error saving payment method into cache: [${paymentMethodResponse.id}]"
+                        )
+                    }
             }
-            .map { paymentMethodResponse }
+            .subscribeOn(Schedulers.parallel())
```

Comment:
I think we can remove this line since we already specify Dispatcher.IO above.

Reply:
Done here b8acb59ccf2e7fbeb22f52ac6789470616a25edf